File: C++/maxArea.cpp
#include<iostream>
#include<vector>

class Solution{
    public:
    int maxArea(std::vector<int>& height) {
        int n =height.size();
        int maxArea =0;
        if(n<2){return 0;}
        int left=0;
        int right=n-1;
        while(left<right){
            int current= std::min(height[left],height[right])*(right-left);
            maxArea=std::max(maxArea,current);
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxArea;
    }
};

int main(){
    std::vector<int>height1={1,7,3,2,4,5,8,2,7};
    std::vector<int>height2={5,4,3,2,1,5};
    Solution solution;
    std::cout<<solution.maxArea(height1)<<"\n"<<solution.maxArea(height2)<<"\n";
    return 0;
}

File: python/maxArea.py
class Solution:
    def maxArea(self , height: list[int]) -> int:
        n=len(height)
        if n<2:
            return 0
        left=0
        right=n-1
        max_area=0
        while left<right:
            current=min(height[left],height[right])*(right-left)
            max_area=max(current,max_area)
            if height[left] <height[right]:
                left +=1
            else:
                right -=1
        return max_area
    
height=[1,7,3,2,4,5,8,2,7]
s=Solution()
print(s.maxArea(height))